File: frontend/app/types/repo.ts
export type RepoCategory = "ML" | "Math" | "Automation" | "Website";

export type Repo = {
  id: number;
  name: string;
  description?: string | null;
  html_url: string;
  language?: string | null;
  stargazers_count?: number | null;
  category?: RepoCategory | null;
};

File: backend/models/repo.py
from pydantic import BaseModel
from typing import Optional,Literal

class Repo(BaseModel):
    id:int
    name:str 
    description: Optional[str]
    html_url:str
    language:Optional[str]=None
    stargazers_count: Optional[int]=0
    category: Optional[Literal["ML","Math","Automation","Website"]]=None